cpp/include/cuspatial/experimental/type_utils.hpp

```diff
@@ -52,20 +86,77 @@ auto make_vec_2d_iterator(FirstIter first, SecondIter second)
   return thrust::make_transform_iterator(zipped, detail::tuple_to_vec_2d<T, VectorType>());
 }
 
+/**
+ * @brief Create an iterator to `lonlat_2d` data from two input iterators.
+ *
+ * Interleaves longitude and latitude from separate iterators into a single iterator to lon/lat
+ * coordinates.
+ * @tparam FirstIter Iterator type to the first component of `lonlat_2d`. Must meet the requirements
```

Comment:
It might be nice to state explicitly that the first component is the longitude and the second is the latitude. I could definitely see users defaulting to think that latitude is "first" based on just colloquial usage of "latitude and longitude".